feat: Added chunk function to paddle frontend

arshPratap opened the pull request:
Close #22321

bipinKrishnan commented on ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py:
```diff
@@ -711,3 +711,53 @@ def test_paddle_rot90(
+
+
+@st.composite
+def _chunk_helper(draw):
```
Is there a need for this helper function, feels like most the things returned by this function can be done with the help of existing helper functions, you could refer to the [hypothesis helper folder](https://github.com/unifyai/ivy/tree/main/ivy_tests/test_ivy/helpers/hypothesis_helpers) or the [testing deep dive](https://unify.ai/docs/ivy/overview/deep_dive/ivy_tests.html).

arshPratap replied:
@bipinKrishnan thanks for the quick review..I will get back with the updates

arshPratap replied:
Resolved @bipinKrishnan

bipinKrishnan commented on ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py:
```diff
@@ -711,3 +711,53 @@ def test_paddle_rot90(
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        input=x[0],
```
The tests are failing due to this unexpected argument `input`, you probably meant `x`.

arshPratap replied:
thanks @bipinKrishnan ..resolved

bipinKrishnan commented on ivy/functional/frontends/paddle/tensor/manipulation.py:
```diff
@@ -44,6 +44,33 @@ def cast(x, dtype):
+)
+@to_ivy_arrays_and_back
+def chunk(x, chunks, axis=0, name=None):
+    if ivy.shape(x) == ():
```
`x` will always be a tensor, right? So we could just do x.shape and also we could raise an error if the shape is `( )` and `chunks>1`.

arshPratap replied:
Resolved @bipinKrishnan !

bipinKrishnan commented on ivy/functional/frontends/paddle/tensor/manipulation.py:
```diff
@@ -44,6 +44,33 @@ def cast(x, dtype):
+def chunk(x, chunks, axis=0, name=None):
+    if ivy.shape(x) == ():
+        return [x]
+    else:
```
No need for the `else` statement here as we already have a `return` above this line.

arshPratap replied:
Resolved @bipinKrishnan

bipinKrishnan commented on ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py:
```diff
@@ -101,6 +101,27 @@ def _broadcast_to_helper(draw):
 
 
+@st.composite
+def _chunk_helper(draw):
```
We could move this function to the helper folder in tests.

arshPratap replied:
Done ! Thanks @bipinKrishnan

bipinKrishnan commented:
Hey @arshPratap , the tests are failing due to ground truths not matching with returned output from the function. Can you please look into this?

bipinKrishnan commented on ivy/functional/frontends/paddle/manipulation.py:
```diff
@@ -44,6 +44,34 @@ def cast(x, dtype):
+@to_ivy_arrays_and_back
+def chunk(x, chunks, axis=0, name=None):
+    if x.shape == ():
+        if chunks > 1:
```
You could join this condition in the line above using `and`

arshPratap replied:
resolved thanks @bipinKrishnan

bipinKrishnan commented on ivy/functional/frontends/paddle/manipulation.py:
```diff
@@ -44,6 +44,34 @@ def cast(x, dtype):
+    chunk_size = axis_size // chunks
+    if chunk_size == 0:
+        return ivy.split(x, num_or_size_splits=axis_size, axis=axis)
+    else:
```
You could remove the else statement since there is a return above. Do the same for rest of the code as well.

bipinKrishnan commented:
@arshPratap The code looks a bit confusing with a lot of return statements. Since `ivy.split` function is the common factor here, can we just use some one liner code with if-else statements to get the `num_or_size_splits`, `axis` and then finally we call `ivy.split` using the obtained values?

arshPratap commented:
> @arshPratap The code looks a bit confusing with a lot of return statements. Since `ivy.split` function is the common factor here, can we just use some one liner code with if-else statements to get the `num_or_size_splits`, `axis` and then finally we call `ivy.split` using the obtained values?

Yup I agree @bipinKrishnan .. have made the changes could you take a look ?

arshPratap commented:
Hi @bipinKrishnan .. could you take a look ?

bipinKrishnan commented on ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:
```diff
@@ -197,27 +200,6 @@ def _arrays_idx_n_dtypes(draw):
-    else:
-        factors = []
-        for i in range(1, shape[axis] + 1):
-            if shape[axis] % i == 0:
```
May be you've rebased incorrectly, these changes shouldn't be shown here. Can you please re-check this?

arshPratap replied:
Hi @bipinKrishnan ..I actually moved this testing function since `torch`  and `paddle`were using the same function and placed the common function into the helper directory.I can rebase and correct it now.No problem

bipinKrishnan replied:
Yeah since this PR is not related to torch, its better to keep the files untouched. Hope that makes sense!

arshPratap replied:
Hi @bipinKrishnan could you take another look ? I have removed the changes that were related to `torch`.

arshPratap commented:
Hi @bipinKrishnan ! Could you take a look  !

Ishticode commented:
resolved conflicts. waiting on ci.

ivy-seed commented:
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

ivy-seed commented:
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
